libvast/include/vast/system/index.hpp

```diff
@@ -117,48 +118,21 @@ class partition_factory {
 
 private:
   filesystem_actor filesystem_;
-  const index_state& state_;
-};
-
-struct query_backlog {
-  struct job {
-    vast::query query;
-    caf::typed_response_promise<query_cursor> rp;
-    caf::actor_addr sender;
-  };
-
-  // Emplace a job.
-  void emplace(vast::query query, caf::typed_response_promise<query_cursor> rp,
-               caf::actor_addr sender);
 
-  /// Cancels jobs associated with the given sender.
-  /// @returns The number of cancelled jobs.
-  uint64_t cancel(caf::actor_addr sender);
-
-  [[nodiscard]] std::optional<job> take_next();
-
-  std::deque<job> normal;
-  std::deque<job> low;
+  // Mutable reference so we can increment the materializations counter.
```

Comment:
Would it be possible to move the counter here instead?

Reply:
Yes that is also possible.